### PR TITLE
Get S3 property from form and set it in S3 client

### PR DESCRIPTION
**Pull request by cparimala**
Updating the plugin to use the Service URL endpoint as per the [ticket](https://cloudbees.atlassian.net/browse/CDRO-608)

### CODE REVIEW

**Comment by jsmith-cb** [comment]
@cparimala,
 
Ensure you add the new version and release note to the help.adoc: [here](https://github.com/electric-cloud/EC-S3/blob/CDRO-608/help/help.adoc?plain=1#L265)

For the release note, something like:
```
=== 1.2.2

* Fixed issue where the plugin configuration *Service URL*  parameter was not applied to procedures. 
```

File: help/help.adoc [hunk 1]
```diff
@@ -264,6 +264,10 @@ image::cloudbees-common-sda::cd-plugins/ec-s3/use-cases/case-1/website-hosting-l
 [[rns]]
 == Amazon Simple Storage Service plugin release notes
 
+=== 1.2.2
+
+* Fixed issue where the plugin configuration *Service URL*  parameter was not applied to procedures.
+
```
Comment (cparimala) [inline help/help.adoc:270]: @jsmith-cb  added release note

File: help/help.adoc [hunk 2]
```diff
@@ -264,6 +264,10 @@ image::cloudbees-common-sda::cd-plugins/ec-s3/use-cases/case-1/website-hosting-l
 [[rns]]
 == Amazon Simple Storage Service plugin release notes
 
+=== 1.2.2
+
+* Fixed issue where the plugin configuration *Service URL*  parameter was not applied to procedures.
```
Comment (jsmith-cb) [inline help/help.adoc:269]:
```suggestion
* Fixed issue where the plugin configuration *Service URL* parameter was not applied to procedures.
```

File: src/main/resources/project/procedures/preamble.groovy [hunk 3]
```diff
@@ -74,60 +73,75 @@ public class ElectricCommander {
...
 
         userName = credential.userName
         password = credential.password
+        try {
+            serviceUrl = getCommanderProperty('service_url')
+            println "service_url: $serviceUrl"
+        } catch (Throwable e) {
+            println "service_url is not found (service_url field does not exist?),setting to default"
+            serviceUrl = "https://s3.amazonaws.com"
+        }
```
Comment (cmarunachalam) [inline src/main/resources/project/procedures/preamble.groovy:103]: Move this logic to separate method and use the return value here.

File: src/main/resources/project/procedures/preamble.groovy [hunk 4]
```diff
@@ -74,60 +73,75 @@ public class ElectricCommander {
...
         password = resp.getData().credential.password
-
+        try {
+            serviceUrl = getCommanderProperty('service_url')
+            println "service_url: $serviceUrl"
+        } catch (Throwable e) {
+            println "service_url is not found (service_url field does not exist?),setting to default"
+            serviceUrl = "https://s3.amazonaws.com"
+        }
     }
```
Comment (cmarunachalam) [inline src/main/resources/project/procedures/preamble.groovy:135]: use the method as I mentioned above and use it here

File: src/main/resources/project/procedures/preamble.groovy (hunk 3 above)
Comment (cparimala) [reply src/main/resources/project/procedures/preamble.groovy]: @cmarunachalam  moved the logic to a new method. And removed the print statements

**Review by cmarunachalam — APPROVED** [review]

**Review by CJkrishnan — APPROVED** [review]

**Pull request merged**